app/mixins/ui-kit/border.js

```diff
@@ -0,0 +1,82 @@
+import Mixin from '@ember/object/mixin';
+import { computed } from '@ember/object';
+import { or, none, not } from '@ember/object/computed';
+import { checkDictionary } from 'travis/utils/ui-kit/assertions';
+
+// Border
+const BORDER_WIDTHS = {
+  NONE: 'none',
+  ONE: '1',
+};
+
+export default Mixin.create({
+  // Public interface //
+  borderColor: null,
+  borderWidth: null,
+
+  // Private //
+  get blank() {
+    return '';
+  },
+
+  hasNoBorderColor: none('borderColor'),
+  hasBorderColor: not('hasNoBorderColor'),
+  borderColorClass: computed('hasNoBorderColor', 'borderColor', function () {
+    return this.hasNoBorderColor ? '' : `border-${this.borderColor}`;
+  }),
+
+  borderTopWidth: or('borderWidth.top', 'borderWidth.y', 'borderWidth.all', 'borderWidth', 'blank'),
+  borderRightWidth: or('borderWidth.right', 'borderWidth.x', 'borderWidth.all', 'borderWidth', 'blank'),
+  borderBottomWidth: or('borderWidth.bottom', 'borderWidth.y', 'borderWidth.all', 'borderWidth', 'blank'),
+  borderLeftWidth: or('borderWidth.left', 'borderWidth.x', 'borderWidth.all', 'borderWidth', 'blank'),
+
+  borderTopWidthClass: computed('borderTopWidth', function () {
+    return this.borderTopWidth.length === 0 ? '' : `border-t-${this.borderTopWidth}`;
+  }),
+  borderRightWidthClass: computed('borderRightWidth', function () {
+    return this.borderRightWidth.length === 0 ? '' : `border-r-${this.borderRightWidth}`;
+  }),
+  borderBottomWidthClass: computed('borderBottomWidth', function () {
+    return this.borderBottomWidth.length === 0 ? '' : `border-b-${this.borderBottomWidth}`;
+  }),
+  borderLeftWidthClass: computed('borderLeftWidth', function () {
+    return this.borderLeftWidth.length === 0 ? '' : `border-l-${this.borderLeftWidth}`;
+  }),
+
+  borderWidthClasses: computed(
+    'borderTopWidthClass',
+    'borderRightWidthClass',
+    'borderBottomWidthClass',
+    'borderLeftWidthClass',
+    function () {
+      return `
+        ${this.borderTopWidthClass}
+        ${this.borderRightWidthClass}
+        ${this.borderBottomWidthClass}
+        ${this.borderLeftWidthClass}
+      `;
+    }
+  ),
+
+  // hasNoBorderWidth: none('borderWidth'),
+  // borderWidthClass: computed('hasNoBorderWidth', 'hasBorderColor', 'borderWidth', function () {
+  //   let currentBorderWidth = this.borderWidth || '';
+  //   if (this.hasNoBorderWidth && this.hasBorderColor) {
+  //     currentBorderWidth = BASE_BORDER_WIDTH;
+  //   }
+  //   return currentBorderWidth.length === 0 ? '' : `border-${currentBorderWidth}`;
+  // }),
```

Comment:
Is it left for a reason?

Reply:
Haha whoops, nah that was accidental. Removed :+1: